### PR TITLE
Support multiple onwards layouts

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR adds support for additional layouts for onwards content based on the number of trails in the array

## Link to supporting Trello card
https://trello.com/c/rBoduy1y/1065-dynamic-onward-layouts

### CODE REVIEW

File: src/web/components/Onwards/ExactlyFive.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
```
Comment (liywjl) [inline src/web/components/Onwards/ExactlyFive.tsx]: I think adding a story for this would be great, to highlight how the 5 cards would be rendered

File: src/web/components/Onwards/OnwardsLayout.tsx [hunk 2]
```diff
@@ -6,41 +6,50 @@ import { Hide } from '@frontend/web/components/Hide';
...
-export const OnwardsLayout = ({ onwardSections }: Props) => {
-    return (
-        <>
-            {onwardSections.map((onward, index) => (
-                <Flex key={`${onward.heading}-${index}`}>
-                    <LeftColumn
-                        showRightBorder={false}
-                        showPartialRightBorder={true}
-                    >
+export const OnwardsLayout = ({ onwardSections }: Props) => (
```
Comment (liywjl) [inline src/web/components/Onwards/OnwardsLayout.tsx]: Optional: unnecessary change, can revert `() => { return (...) }` back to `() => (...)`

Comment (oliverlloyd) [reply src/web/components/Onwards/OnwardsLayout.tsx]: Ah, yes, good idea!

File: src/web/components/Onwards/ExactlyFive.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    content: TrailType[];
+};
+
+export const ExactlyFive = ({ content }: Props) => (
+    <>
+        <UL direction="row" bottomMargin={true}>
+            <LI padSides={true} percentage="33%">
+                <Card
+                    {...{
+                        linkTo: content[0].url,
```
Comment (gtrufitt) [inline src/web/components/Onwards/ExactlyFive.tsx]: I assume `content[0]` is typesafe?

File: src/web/components/Onwards/ExactlyFive.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            </LI>
+            <LI
+                padSides={true}
+                showDivider={true}
+                showTopMarginWhenStacked={true}
+                percentage="33%"
+            >
+                <Card
+                    {...{
+                        linkTo: content[1].url,
```
Comment (gtrufitt) [inline src/web/components/Onwards/ExactlyFive.tsx]: As you repeat this a lot, maybe it could be  a func?

```js
const getCardProps = (content, storyN) => {
  const c = content[storyN];
  return ({
    linkTo: c.url
  })
}
```

Would at least save having to go through each if you want to add/remove a prop. And also make it easier to see the structure of the Components!

File: src/web/components/Onwards/ExactlyFive.tsx (hunk 3 above)
Comment (oliverlloyd) [reply src/web/components/Onwards/ExactlyFive.tsx]: Yes, should be. Typescript knows what properties should be available on that array object.

File: src/web/components/Onwards/ExactlyFive.tsx (hunk 4 above)
Comment (oliverlloyd) [reply src/web/components/Onwards/ExactlyFive.tsx]: There are some subtle changes in which properties get called for different cards which could make an abstraction more complicated. Eg. sometimes image isn't included and headlineSize can vary.

If we had exact parity on property names we could get away with spreading the content object and applying any changes afterwards, like:

```
                <Card
                    {...{
                        ...content[0],
                        image: '',
                    }}
                />
```

but we're out of sync on a couple of things so we'd need to be explicit for those

Comment (gtrufitt) [reply src/web/components/Onwards/ExactlyFive.tsx]: Ah ok, I think that'd be a nice enhancement to align those - easier to reason about probably and would make this much cleaner. Future refactor

Comment (oliverlloyd) [reply src/web/components/Onwards/ExactlyFive.tsx]: Yes, defo. It's kind of intentionally verbose at the moment with the goal of finding a pattern once we understand the different ways we use it
